### PR TITLE
Handle stripe webhook event verifying signature

### PR DESCRIPTION
**Pull request by hdiniz**

### CODE REVIEW

**Comment by kewitz** [comment]
Related to this: https://github.com/opencollective/opencollective/issues/6203

**Comment by znarf** [comment]
@kewitz What's your review? Are your good with this change?

Generally, any concern that I should be aware if we were merging this?

File: server/models/VirtualCard.ts [hunk 1]
```diff
@@ -27,7 +35,8 @@ class VirtualCard extends Model<InferAttributes<VirtualCard, { omit: 'info' }>,
 
   // Associations
   public declare collective?: NonAttribute<any>;
-  public declare host?: NonAttribute<any>;
+  public declare host?: NonAttribute<typeof Collective>;
+  public declare getHost: BelongsToGetAssociationMixin<typeof Collective>;
```
Comment (kewitz) [inline server/models/VirtualCard.ts:39]: 💯

File: server/paymentProviders/stripe/virtual-cards.ts [hunk 2]
```diff
@@ -104,14 +108,16 @@ export const updateVirtualCardLimit = async (
       spending_limits: [
         {
           amount: limitAmount,
-          interval: limitInterval.toLowerCase(),
+          interval: <Stripe.Issuing.CardUpdateParams.SpendingControls.SpendingLimit.Interval>(
```
Comment (kewitz) [inline server/paymentProviders/stripe/virtual-cards.ts:111]: Puts the Java in JavaScript. 😂

File: server/paymentProviders/stripe/webhook.ts [hunk 3]
```diff
@@ -564,78 +563,92 @@ async function paymentMethodAttached(event: Stripe.Response<Stripe.Event>) {
...
-    signature: request.headers['stripe-signature'],
-    rawBody: request.rawBody,
-  };
+  let webhookSigningSecret =
+    stripeConnectedAccount.data?.webhookSigningSecret || stripeConnectedAccount.data.stripeEndpointSecret;
 
-  if (requestBody.type === 'issuing_authorization.request') {
-    return virtualcard.processAuthorization(requestBody.data.object, stripeEvent);
+  if (process.env.NODE_ENV === 'development' && !webhookSigningSecret && process.env.STRIPE_WEBHOOK_SIGNING_SECRET) {
+    webhookSigningSecret = process.env.STRIPE_WEBHOOK_SIGNING_SECRET;
```
Comment (kewitz) [inline server/paymentProviders/stripe/webhook.ts:588]: Ideally, we would move all events to the platform account and use only the signing secret in our env.
We can merge this and work on https://github.com/opencollective/opencollective/issues/6203 later.

**Review by kewitz — APPROVED** [review]
Awesome work!

**Pull request merged**